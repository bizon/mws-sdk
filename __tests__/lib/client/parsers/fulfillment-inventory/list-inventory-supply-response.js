const parseXml = require('../../../../../lib/client/parsers')
const parseListInventorySupplyResponse = require('../../../../../lib/client/parsers/fulfillment-inventory/list-inventory-supply-response')

describe('lib.client.parsers.fulfillment-inventory.list-inventory-supply-response', () => {
  it('should parse the ListInventorySupply example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListInventorySupplyResponse xmlns="http://mws.amazonaws.com/FulfillmentInventory/2010-10-01/">
        <ListInventorySupplyResult>
          <InventorySupplyList>
            <member>
              <SellerSKU>SampleSKU1</SellerSKU>
              <ASIN>B00000K3CQ</ASIN>
              <TotalSupplyQuantity>20</TotalSupplyQuantity>
              <FNSKU>X0000000FM</FNSKU>
              <Condition>NewItem</Condition>
              <SupplyDetail />
              <InStockSupplyQuantity>15</InStockSupplyQuantity>
              <EarliestAvailability>
                <TimepointType>Immediately</TimepointType>
              </EarliestAvailability>
            </member>
            <member>
              <SellerSKU>SampleSKU2</SellerSKU>
              <ASIN>B00004RWQR</ASIN>
              <TotalSupplyQuantity>0</TotalSupplyQuantity>
              <FNSKU>X00008FZR1</FNSKU>
              <Condition>UsedLikeNew</Condition>
              <SupplyDetail />
              <InStockSupplyQuantity>0</InStockSupplyQuantity>
            </member>
          </InventorySupplyList>
        </ListInventorySupplyResult>
        <ResponseMetadata>
          <RequestId>e8698ffa-8e59-11df-9acb-230ae7a8b736</RequestId>
        </ResponseMetadata>
      </ListInventorySupplyResponse>`,
    )

    const response = parseListInventorySupplyResponse(
      '/fulfillmentInventory:ListInventorySupplyResponse',
      doc,
    )

    expect(response).toMatchSnapshot()
  })

  it('should parse a ListInventorySupplyResponse with a MarketplaceId', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListInventorySupplyResponse xmlns="http://mws.amazonaws.com/FulfillmentInventory/2010-10-01/">
        <ListInventorySupplyResult>
          <MarketplaceId>A13V1IB3VIYZZH</MarketplaceId>
          <InventorySupplyList>
            <member>
              <Condition>NewItem</Condition>
              <SupplyDetail>
                <member>
                  <LatestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </LatestAvailableToPick>
                  <EarliestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </EarliestAvailableToPick>
                  <Quantity>11</Quantity>
                  <SupplyType>InStock</SupplyType>
                </member>
                <member>
                  <LatestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </LatestAvailableToPick>
                  <EarliestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </EarliestAvailableToPick>
                  <Quantity>16</Quantity>
                  <SupplyType>InStock</SupplyType>
                </member>
                <member>
                  <LatestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </LatestAvailableToPick>
                  <EarliestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </EarliestAvailableToPick>
                  <Quantity>1</Quantity>
                  <SupplyType>InStock</SupplyType>
                </member>
              </SupplyDetail>
              <TotalSupplyQuantity>28</TotalSupplyQuantity>
              <EarliestAvailability>
                <TimepointType>Immediately</TimepointType>
              </EarliestAvailability>
              <FNSKU>X00008FZR1</FNSKU>
              <InStockSupplyQuantity>28</InStockSupplyQuantity>
              <ASIN>B00004RWQR</ASIN>
              <SellerSKU>SampleSKU1</SellerSKU>
            </member>
          </InventorySupplyList>
        </ListInventorySupplyResult>
        <ResponseMetadata>
          <RequestId>8734426a-a50f-4c8e-a3cd-d8d39e532d76</RequestId>
        </ResponseMetadata>
      </ListInventorySupplyResponse>`,
    )

    const response = parseListInventorySupplyResponse(
      '/fulfillmentInventory:ListInventorySupplyResponse',
      doc,
    )

    expect(response).toMatchSnapshot()
  })

  it('should parse the ListInventorySupplyByNextToken example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListInventorySupplyByNextTokenResponse xmlns="http://mws.amazonaws.com/FulfillmentInventory/2010-10-01/">
        <ListInventorySupplyByNextTokenResult>
          <NextToken>2YgYW55IGNhcm5hbCBwbGVhc3VyZS4=</NextToken>
          <InventorySupplyList>
            <member>
              <SellerSKU>SampleSKU5</SellerSKU>
              <ASIN>B00008UI0R</ASIN>
              <TotalSupplyQuantity>0</TotalSupplyQuantity>
              <FNSKU>B00008UI0R</FNSKU>
              <Condition />
              <SupplyDetail />
              <InStockSupplyQuantity>0</InStockSupplyQuantity>
            </member>
            <member>
              <SellerSKU>SampleSKU6</SellerSKU>
              <ASIN>B00000K3CQ</ASIN>
              <TotalSupplyQuantity>5259</TotalSupplyQuantity>
              <FNSKU>X0000000FM</FNSKU>
              <Condition>NewItem</Condition>
              <SupplyDetail />
              <InStockSupplyQuantity>5259</InStockSupplyQuantity>
              <EarliestAvailability>
                <TimepointType>Immediately</TimepointType>
              </EarliestAvailability>
            </member>
          </InventorySupplyList>
        </ListInventorySupplyByNextTokenResult>
        <ResponseMetadata>
          <RequestId>b3a020ac-8e5e-11df-9acb-230ae7a8b736</RequestId>
        </ResponseMetadata>
      </ListInventorySupplyByNextTokenResponse>`,
    )

    const response = parseListInventorySupplyResponse(
      '/fulfillmentInventory:ListInventorySupplyByNextTokenResponse',
      doc,
      true,
    )

    expect(response).toMatchSnapshot()
  })

  it('should parse a ListInventorySupplyByNextToken with a MarketplaceId', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListInventorySupplyByNextTokenResponse xmlns="http://mws.amazonaws.com/FulfillmentInventory/2010-10-01/">
        <ListInventorySupplyByNextTokenResult>
          <MarketplaceId>A13V1IB3VIYZZH</MarketplaceId>
          <InventorySupplyList>
            <member>
              <Condition>NewItem</Condition>
              <SupplyDetail>
                <member>
                  <LatestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </LatestAvailableToPick>
                  <EarliestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </EarliestAvailableToPick>
                  <Quantity>11</Quantity>
                  <SupplyType>InStock</SupplyType>
                </member>
                <member>
                  <LatestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </LatestAvailableToPick>
                  <EarliestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </EarliestAvailableToPick>
                  <Quantity>16</Quantity>
                  <SupplyType>InStock</SupplyType>
                </member>
                <member>
                  <LatestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </LatestAvailableToPick>
                  <EarliestAvailableToPick>
                    <TimepointType>Immediately</TimepointType>
                  </EarliestAvailableToPick>
                  <Quantity>1</Quantity>
                  <SupplyType>InStock</SupplyType>
                </member>
              </SupplyDetail>
              <TotalSupplyQuantity>28</TotalSupplyQuantity>
              <EarliestAvailability>
                <TimepointType>Immediately</TimepointType>
              </EarliestAvailability>
              <FNSKU>X00008FZR1</FNSKU>
              <InStockSupplyQuantity>28</InStockSupplyQuantity>
              <ASIN>B00004RWQR</ASIN>
              <SellerSKU>SampleSKU1</SellerSKU>
            </member>
          </InventorySupplyList>
        </ListInventorySupplyByNextTokenResult>
        <ResponseMetadata>
          <RequestId>8734426a-a50f-4c8e-a3cd-d8d39e532d76</RequestId>
        </ResponseMetadata>
      </ListInventorySupplyByNextTokenResponse>`,
    )

    const response = parseListInventorySupplyResponse(
      '/fulfillmentInventory:ListInventorySupplyByNextTokenResponse',
      doc,
      true,
    )

    expect(response).toMatchSnapshot()
  })
})
