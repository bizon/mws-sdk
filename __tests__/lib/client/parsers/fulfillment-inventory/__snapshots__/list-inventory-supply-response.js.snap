// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.fulfillment-inventory.list-inventory-supply-response should parse a ListInventorySupplyByNextToken with a MarketplaceId 1`] = `
{
  "listInventorySupplyResult": {
    "inventorySupplyList": [
      {
        "asin": "B00004RWQR",
        "condition": "NewItem",
        "earliestAvailability": {
          "dateTime": null,
          "timepointType": "Immediately",
        },
        "fnSKU": "X00008FZR1",
        "inStockSupplyQuantity": 28,
        "sellerSKU": "SampleSKU1",
        "supplyDetail": [
          {
            "earliestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "latestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "quantity": 11,
            "supplyType": "InStock",
          },
          {
            "earliestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "latestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "quantity": 16,
            "supplyType": "InStock",
          },
          {
            "earliestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "latestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "quantity": 1,
            "supplyType": "InStock",
          },
        ],
        "totalSupplyQuantity": 28,
      },
    ],
    "marketplaceId": "A13V1IB3VIYZZH",
    "nextToken": null,
  },
  "responseMetadata": {
    "requestId": "8734426a-a50f-4c8e-a3cd-d8d39e532d76",
  },
}
`;

exports[`lib.client.parsers.fulfillment-inventory.list-inventory-supply-response should parse a ListInventorySupplyResponse with a MarketplaceId 1`] = `
{
  "listInventorySupplyResult": {
    "inventorySupplyList": [
      {
        "asin": "B00004RWQR",
        "condition": "NewItem",
        "earliestAvailability": {
          "dateTime": null,
          "timepointType": "Immediately",
        },
        "fnSKU": "X00008FZR1",
        "inStockSupplyQuantity": 28,
        "sellerSKU": "SampleSKU1",
        "supplyDetail": [
          {
            "earliestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "latestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "quantity": 11,
            "supplyType": "InStock",
          },
          {
            "earliestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "latestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "quantity": 16,
            "supplyType": "InStock",
          },
          {
            "earliestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "latestAvailableToPick": {
              "dateTime": null,
              "timepointType": "Immediately",
            },
            "quantity": 1,
            "supplyType": "InStock",
          },
        ],
        "totalSupplyQuantity": 28,
      },
    ],
    "marketplaceId": "A13V1IB3VIYZZH",
    "nextToken": null,
  },
  "responseMetadata": {
    "requestId": "8734426a-a50f-4c8e-a3cd-d8d39e532d76",
  },
}
`;

exports[`lib.client.parsers.fulfillment-inventory.list-inventory-supply-response should parse the ListInventorySupply example response from MWS doc 1`] = `
{
  "listInventorySupplyResult": {
    "inventorySupplyList": [
      {
        "asin": "B00000K3CQ",
        "condition": "NewItem",
        "earliestAvailability": {
          "dateTime": null,
          "timepointType": "Immediately",
        },
        "fnSKU": "X0000000FM",
        "inStockSupplyQuantity": 15,
        "sellerSKU": "SampleSKU1",
        "supplyDetail": [],
        "totalSupplyQuantity": 20,
      },
      {
        "asin": "B00004RWQR",
        "condition": "UsedLikeNew",
        "earliestAvailability": null,
        "fnSKU": "X00008FZR1",
        "inStockSupplyQuantity": 0,
        "sellerSKU": "SampleSKU2",
        "supplyDetail": [],
        "totalSupplyQuantity": 0,
      },
    ],
    "marketplaceId": null,
    "nextToken": null,
  },
  "responseMetadata": {
    "requestId": "e8698ffa-8e59-11df-9acb-230ae7a8b736",
  },
}
`;

exports[`lib.client.parsers.fulfillment-inventory.list-inventory-supply-response should parse the ListInventorySupplyByNextToken example response from MWS doc 1`] = `
{
  "listInventorySupplyResult": {
    "inventorySupplyList": [
      {
        "asin": "B00008UI0R",
        "condition": "",
        "earliestAvailability": null,
        "fnSKU": "B00008UI0R",
        "inStockSupplyQuantity": 0,
        "sellerSKU": "SampleSKU5",
        "supplyDetail": [],
        "totalSupplyQuantity": 0,
      },
      {
        "asin": "B00000K3CQ",
        "condition": "NewItem",
        "earliestAvailability": {
          "dateTime": null,
          "timepointType": "Immediately",
        },
        "fnSKU": "X0000000FM",
        "inStockSupplyQuantity": 5259,
        "sellerSKU": "SampleSKU6",
        "supplyDetail": [],
        "totalSupplyQuantity": 5259,
      },
    ],
    "marketplaceId": null,
    "nextToken": "2YgYW55IGNhcm5hbCBwbGVhc3VyZS4=",
  },
  "responseMetadata": {
    "requestId": "b3a020ac-8e5e-11df-9acb-230ae7a8b736",
  },
}
`;
